applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1117,6 +1119,8 @@ static void disconnect(const void *subscriber_id, u8_t report_id)
 
 	rs->subscriber = NULL;
 	rs->state = STATE_DISCONNECTED;
+	__ASSERT_NO_MSG(subscriber->report_cnt >= rs->cnt);
```

Comment:
Here we would have to round robin the reports to see if no report was blocked on the subscriber and so hid state can send its data.
I know that for current implementation of the USB we disable all reports at once but let's keep this logically correct.

As discussed offline it would be better to simply send a report sent event with an error when USB disconnects.